ddl/partition.go

```diff
@@ -576,38 +576,48 @@ func validRangePartitionType(col *model.ColumnInfo) bool {
 }
 
 // checkDropTablePartition checks if the partition exists and does not allow deleting the last existing partition in the table.
-func checkDropTablePartition(meta *model.TableInfo, partName string) error {
+func checkDropTablePartition(meta *model.TableInfo, partNames []string) error {
 	pi := meta.Partition
 	if pi.Type != model.PartitionTypeRange && pi.Type != model.PartitionTypeList {
 		return errOnlyOnRangeListPartition.GenWithStackByArgs("DROP")
 	}
 	oldDefs := pi.Definitions
-	for _, def := range oldDefs {
-		if strings.EqualFold(def.Name.L, strings.ToLower(partName)) {
-			if len(oldDefs) == 1 {
-				return errors.Trace(ErrDropLastPartition)
+	for _, pn := range partNames {
+		found := false
+		for _, def := range oldDefs {
+			if strings.EqualFold(def.Name.L, strings.ToLower(pn)) {
```

Comment:
ok it's not introduced by this PR, why do we do `EqualFold(L, ToLower(pn))` 😱 either the EqualFold or the ToLower should be removed

```suggestion
			if def.Name.L == strings.ToLower(pn) {
```